src/graphforge/datasets/registry.py

```
@@ -0,0 +1,252 @@
+def register_loader(format_name: str, loader_class: type[DatasetLoader]) -> None:
+    """Register a dataset loader for a specific format.
+
+    Args:
+        format_name: Format identifier (e.g., "csv", "cypher", "graphml")
+        loader_class: The DatasetLoader class to use for this format
+
+    Raises:
+        ValueError: If a loader for this format is already registered
+    """
+    if format_name in _LOADER_REGISTRY:
+        raise ValueError(f"Loader for format '{format_name}' is already registered")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)